### PR TITLE
http: idle timeout cleanup

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes few nits noted in this PR https://github.com/envoyproxy/envoy/pull/2714
Description: Fixes few nits noted in this PR https://github.com/envoyproxy/envoy/pull/2714. More of cosmetic changes. No functional change.
Risk Level: Low. More of formatting related changes.
Testing: Automated tests
Docs Changes:  NA
Release Notes: NA

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@mattklein123 addressed the review comments of https://github.com/envoyproxy/envoy/pull/2714. PTAL.

**Comment by zuercher** [comment]
@ramaraochavali can you fill out the PR description using the [pull request template](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUEST_TEMPLATE.md).

File: test/integration/http2_integration_test.cc [hunk 1]
```diff
@@ -221,7 +221,7 @@ TEST_P(Http2IntegrationTest, IdleTimeoutWithSimultaneousRequests) {
     auto* cluster = static_resources->mutable_clusters(0);
     auto* http_protocol_options = cluster->mutable_common_http_protocol_options();
     auto* idle_time_out = http_protocol_options->mutable_idle_timeout();
-    std::chrono::milliseconds timeout(1000);
+    std::chrono::milliseconds timeout(3000);
```
Comment (mattklein123) [inline test/integration/http2_integration_test.cc]: Why the changes here? Is there some other underlying timing issue with the test?

Comment (ramaraochavali) [reply test/integration/http2_integration_test.cc]: @mattklein123 the test failed only on Mac once with not getting response for second request. I suspected that it could be because of idletimeout and increased it. But looking at the in-flight PRs it seems to be working fine. So I will revert this and we can revisit if it causes problem again.

File: test/common/http/codec_client_test.cc [hunk 2]
```diff
@@ -158,6 +158,7 @@ TEST_F(CodecClientTest, IdleTimerWithNoActiveRequests) {
   // Close the client and validate idleTimer is reset
   EXPECT_EQ(client_->numActiveRequests(), 0);
   client_->close();
+  // TODO: Use default connection mock handlers for raising events.
```
Comment (zuercher) [inline test/common/http/codec_client_test.cc]: Per the style guide, please put your github user name in these TODOs: `//TODO(ramaraochavali): ...`
